src/files.rs

```diff
@@ -0,0 +1,173 @@
+use super::ApiResponseOrError;
+use crate::{openai_delete, openai_get, openai_post_multipart, OpenAiError};
+use derive_builder::Builder;
+use reqwest::multipart::{Form, Part};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct File {
+    pub id: String,
+    pub object: String,
+    pub bytes: usize,
+    pub created_at: usize,
+    pub filename: String,
+    pub purpose: String,
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct DeletedFile {
+    pub id: String,
+    pub object: String,
+    pub deleted: bool,
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct Files {
+    data: Vec<File>,
+    object: String,
+}
+
+#[derive(Serialize, Builder, Debug, Clone)]
+#[builder(pattern = "owned")]
+#[builder(name = "FileUploadBuilder")]
+#[builder(setter(strip_option, into))]
+pub struct FileUploadRequest {
+    filename: String,
```

Comment:
Could this be renamed to `file_name` instead, for consistency? Same for the `filepath` instances.